front/lib/resources/membership_resource.ts

```diff
@@ -0,0 +1,344 @@
+  extends ReadonlyAttributesType<MembershipModel> {}
+export class MembershipResource extends BaseResource<MembershipModel> {
+  static model: ModelStatic<MembershipModel> = MembershipModel;
+
+  constructor(
+    model: ModelStatic<MembershipModel>,
+    blob: Attributes<MembershipModel>
+  ) {
+    super(MembershipModel, blob);
+  }
+
+  static async getActiveMemberships({
+    userIds,
+    workspace,
+    roles,
+  }: GetMembershipsOptions): Promise<MembershipResource[]> {
+    const whereClause: WhereOptions<InferAttributes<MembershipModel>> = {
+      startAt: {
+        [Op.lte]: new Date(),
+      },
+      endAt: {
+        [Op.or]: [{ [Op.eq]: null }, { [Op.gte]: new Date() }],
+      },
+    };
+
+    if (userIds) {
+      whereClause.userId = userIds;
+    }
+    if (workspace) {
+      whereClause.workspaceId = workspace.id;
+    }
+    if (roles) {
+      whereClause.role = {
+        [Op.in]: roles,
+      };
+    }
+
+    const memberships = await MembershipModel.findAll({
+      where: whereClause,
+    });
+
+    return memberships.map(
+      (membership) => new MembershipResource(MembershipModel, membership.get())
+    );
+  }
+
+  static async getLatestMemberships({
+    userIds,
+    workspace,
+    roles,
+  }: GetMembershipsOptions): Promise<MembershipResource[]> {
+    const orderedResourcesFromModels = (resources: MembershipModel[]) =>
+      resources
+        .sort((a, b) => a.startAt.getTime() - b.startAt.getTime())
+        .map(
+          (resource) => new MembershipResource(MembershipModel, resource.get())
+        );
+
+    const where: WhereOptions<InferAttributes<MembershipModel>> = {};
+    if (roles) {
+      where.role = {
+        [Op.in]: roles,
+      };
+    }
+    if (userIds) {
+      where.userId = {
+        [Op.in]: userIds,
+      };
+    }
+    if (workspace) {
+      where.workspaceId = workspace.id;
+    }
+
+    if (!workspace && !userIds?.length) {
+      throw new Error("At least one of workspace or userIds must be provided.");
+    }
+    if (userIds && !userIds.length) return [];
+
+    // Get all the memberships matching the criteria.
+    const memberships = await MembershipModel.findAll({
+      where,
+      order: [["startAt", "DESC"]],
+    });
+    // Then, we only keep the latest membership for each (user, workspace).
+    const latestMembershipByUserAndWorkspace = new Map<
+      string,
+      MembershipModel
+    >();
+    for (const m of memberships) {
+      const key = `${m.userId}__${m.workspaceId}`;
+      const latest = latestMembershipByUserAndWorkspace.get(key);
+      if (!latest || latest.startAt < m.startAt) {
+        latestMembershipByUserAndWorkspace.set(key, m);
+      }
+    }
+
+    return orderedResourcesFromModels(
+      Array.from(latestMembershipByUserAndWorkspace.values())
+    );
+  }
+
+  static async getLatestMembershipOfUserInWorkspace({
+    userId,
+    workspace,
+  }: {
+    userId: number;
+    workspace: WorkspaceType | Workspace;
```

Comment:
Let's only accept LightWorkspaceType here and add a util function in api/workspace to get a WorkspaceType from a Workspace.